shared/slices/slice_generic.go

```diff
@@ -0,0 +1,153 @@
+package slices
+
+import (
+	"fmt"
+	"reflect"
+)
+
+func interfaceToSlice(slice interface{}) ([]interface{}, error) {
+	s := reflect.ValueOf(slice)
+	if s.Kind() != reflect.Slice {
+		return nil, fmt.Errorf("slice error: not of type slice")
+	}
+	ret := make([]interface{}, s.Len())
+	for i := 0; i < s.Len(); i++ {
+		ret[i] = s.Index(i).Interface()
+	}
+	return ret, nil
+}
+
+// GenericIntersection returns a new set with elements that are common in
+// both sets a and b.
+func GenericIntersection(a, b interface{}) (reflect.Value, error) {
+
+	set := reflect.MakeSlice(reflect.TypeOf(a), 0, 0)
+	set1, err1 := interfaceToSlice(a)
```

Comment:
You should handle each error individually, otherwise
when looking through the error logs you will not know what the error is 
```go
set1, err1 := interfaceToSlice(a)
if err1 != nil {
return set, fmt.Errorf("slice type is invalid %v",err1)
}

set2, err2 := interfaceToSlice(a)
if err2 != nil {
return set, fmt.Errorf("slice type is invalid %v",err2)
}
```
The same applies for the other functions